instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -190,6 +168,33 @@ def _determine_parent_context(
     return parent_context
 
 
+def _determine_links() -> Optional[Sequence[Link]]:
+    """Determine if a Link should be added to the Span based on the
+    environment variable `_X_AMZN_TRACE_ID`.
+
+
+    See more:
+    https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#aws-x-ray-environment-span-link
+
+    Returns:
+        A Link or None
+    """
+    links = None
+
+    xray_env_var = os.environ.get(_X_AMZN_TRACE_ID)
+
+    if xray_env_var:
+        env_context = AwsXRayPropagator().extract(
+            {TRACE_HEADER_KEY: xray_env_var}
+        )
+
+        span_context = get_current_span(env_context).get_span_context()
+        if span_context.is_valid:
+            links = [Link(span_context)]
```

Comment:
The link should have a `source=x-ray-env` attribute per the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/7c2312534b539fc599fd1f0901e874c6e489c37a/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#aws-x-ray-environment-span-link).